.github/workflows/ci.yml

```diff
@@ -15,9 +14,13 @@ concurrency:
   cancel-in-progress: ${{ (github.ref == 'refs/heads/staging') || (github.ref == 'refs/heads/trying') }}
 
 jobs:
-  test:
+  integration-tests:
     runs-on: ${{ matrix.os }}
     timeout-minutes: 60
+    # We don't actually want to run integration tests on pull requests,
+    # because we want to avoid hitting rate limits.
+    # So, if this is a PR build, mark the integration tests as "skipped".
+    if: github.event_name != 'pull_request'
```

Comment:
Do I understand correctly this is a required job and so skipping it is making all PRs unmergeable? That's at least my explanation for this PR waiting for a job which will never run: https://github.com/JuliaLang/BumpStdlibs.jl/queue/master

Reply:
> Do I understand correctly this is a required job and so skipping it is making all PRs unmergeable? That's at least my explanation for this PR waiting for a job which will never run: https://github.com/JuliaLang/BumpStdlibs.jl/queue/master

GitHub has a very weird rule: if a required job is marked as "skipped", then GitHub considers it as having satisfied the requirement.

E.g. on this PR, GitHub says "all checks have passed".

Reply:
Yeah, because you aren't running much.  But this job is ***necessary*** for _merging_, in fact this PR [was removed from the merge queue](https://github.com/JuliaLang/BumpStdlibs.jl/pull/110#event-15816132128) in the meantime.  I see two options:

* don't make this job mandatory
* don't skip it for PRs

Reply:
> But this job is _**necessary**_ for _merging_, in fact this PR [was removed from the merge queue](https://github.com/JuliaLang/BumpStdlibs.jl/pull/110#event-15816132128) in the meantime. I see two options:

So, the overall workflow (`CI`) needs to start before GitHub marks this job as skipped. That can't happen until this PR takes the lock. So what happened was just that it took a long time to take the lock, and the merge queue timeout (the separate timeout that lives in the repo settings) was exceeded.

I've increased that timeout.

If we set the merge queue timeout to be really high (

To clarify again: this PR can absolutely be merged with a status of "skipped" for the integration tests.

Reply:
In other words, with GitHub Merge Queue, it is very much allowed to both (1) Make a job required, and (1) mark it as "skipped" on PRs. We currently do this exact thing on RegistryCI.jl and CompatHelper.jl.

But yes, because of the global lock, you often have to wait a very long time to get the lock, so I usually set the merge queue timeout to the max allowed timeout (6 hours).